java-generator/core/src/main/java/io/fabric8/java/generator/nodes/AbstractJSONSchema2Pojo.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.java.generator.nodes;
+
+import static io.fabric8.java.generator.nodes.Keywords.JAVA_KEYWORDS;
+
+import com.github.javaparser.ast.CompilationUnit;
+import io.fabric8.java.generator.exceptions.JavaGeneratorException;
+import io.fabric8.kubernetes.api.model.apiextensions.v1.JSONSchemaProps;
+import java.util.function.Function;
+
+public abstract class AbstractJSONSchema2Pojo {
+
+    static final String BOOLEAN_CRD_TYPE = "boolean";
+    static final String INTEGER_CRD_TYPE = "integer";
+    static final String INT32_CRD_TYPE = "int32";
+    static final String INT64_CRD_TYPE = "int64";
+    static final String NUMBER_CRD_TYPE = "number";
+    static final String FLOAT_CRD_TYPE = "float";
+    static final String DOUBLE_CRD_TYPE = "double";
+    static final String STRING_CRD_TYPE = "string";
+    static final String OBJECT_CRD_TYPE = "object";
+    static final String ARRAY_CRD_TYPE = "array";
```

Comment:
Is there a reason this isn't an enum?

Reply:
not sure I understand the question ...
An `array` is identified by the relevant keyword, instead, an `enum` is defined by the fact that it contains an `enum` field.

Reply:
This whole list of types fields seems like it can be an enum. I saw them used with a switch statement on a string

Reply:
This implementation is somehow similar to the one in the CRD generator:
https://github.com/fabric8io/kubernetes-client/blob/f409dcf256ffbcb2dd5b9b6ec2bf5c5c0d8c8f67/crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java#L67